browser/ui/sidebar/sidebar_browsertest.cc

```diff
@@ -756,6 +795,126 @@ IN_PROC_BROWSER_TEST_F(SidebarBrowserTest, DisabledItemsTest) {
   }
 }
 
+class MockSidebarModelObserver : public SidebarModel::Observer {
+ public:
+  MockSidebarModelObserver() = default;
+  ~MockSidebarModelObserver() override = default;
+
+  MOCK_METHOD(void,
+              OnItemAdded,
+              (const SidebarItem& item, size_t index, bool user_gesture),
+              (override));
+  MOCK_METHOD(void,
+              OnItemMoved,
+              (const SidebarItem& item, size_t from, size_t to),
+              (override));
+  MOCK_METHOD(void, OnItemRemoved, (size_t index), (override));
+  MOCK_METHOD(void,
+              OnActiveIndexChanged,
+              (std::optional<size_t> old_index,
+               std::optional<size_t> new_index),
+              (override));
+  MOCK_METHOD(void,
+              OnItemUpdated,
+              (const SidebarItem& item, const SidebarItemUpdate& update),
+              (override));
+  MOCK_METHOD(void,
+              OnFaviconUpdatedForItem,
+              (const SidebarItem& item, const gfx::ImageSkia& image),
+              (override));
+};
+
+class SidebarBrowserTestWithkSidebarShowAlwaysOnStable
+    : public testing::WithParamInterface<bool>,
+      public SidebarBrowserTest {
+ public:
+  SidebarBrowserTestWithkSidebarShowAlwaysOnStable() {
+    if (GetParam()) {
+      feature_list_.InitAndEnableFeatureWithParameters(
+          sidebar::features::kSidebarShowAlwaysOnStable,
+          {{"open_one_shot_leo_panel", "true"}});
+    } else {
+      feature_list_.InitAndEnableFeature(
+          sidebar::features::kSidebarShowAlwaysOnStable);
+    }
+  }
+  ~SidebarBrowserTestWithkSidebarShowAlwaysOnStable() override = default;
+
+  void SetUp() override { SidebarBrowserTest::SetUp(); }
+
+  void TearDown() override { SidebarBrowserTest::TearDown(); }
+
+  // For skipping SidebarBrowserTest's PreRunTestOnMainThread
+  // as show option is set explicitely there.
+  void PreRunTestOnMainThread() override {
+    InProcessBrowserTest::PreRunTestOnMainThread();
+  }
+
+  void SetUpCommandLine(base::CommandLine* command_line) override {
+    SidebarBrowserTest::SetUpCommandLine(command_line);
+    command_line->AppendSwitch(switches::kDontShowAlwaysSidebarOnNonStable);
+    command_line->AppendSwitch(switches::kForceFirstRun);
+  }
+
+  base::test::ScopedFeatureList feature_list_;
+  testing::NiceMock<MockSidebarModelObserver> observer_;
+  base::ScopedObservation<SidebarModel, SidebarModel::Observer> observation_{
+      &observer_};
+};
+
+IN_PROC_BROWSER_TEST_P(SidebarBrowserTestWithkSidebarShowAlwaysOnStable,
+                       SidebarShowAlwaysTest) {
+  observation_.Observe(model());
+
+  auto* sidebar_service =
+      SidebarServiceFactory::GetForProfile(browser()->profile());
+  EXPECT_EQ(SidebarService::ShowSidebarOption::kShowAlways,
+            sidebar_service->GetSidebarShowOption());
+
+  // Check one shot leo panel is opened or not based on test parameter.
```

Comment:
Maybe leo => Leo? other places ditto

```suggestion
  // Check one shot Leo panel is opened or not based on test parameter.
```

Reply:
Updated all.